### PR TITLE
Contributing docs

### PR DESCRIPTION
**Pull request by gnufied**

### CODE REVIEW

**Comment by tsheasha** [comment]
The part about failing build is not only for contributing to fullerite, the build fails for anyone that wants to install vanilla fullerite without any contributions. I think the audience of the build failure should be to all the users not just the ones who want to contribute.

**Comment by gnufied** [comment]
I see what you are saying T. A new person cloning/downloading the code, won't see the error though.

What you think, we should do? We can automate this via Makefile but I am hesitant because going ahead only those who have an old build/checkout of fullerite will see the error and 99% of our users will not run into this problem.

We can move this into main README.md.

**Comment by tsheasha** [comment]
I don't think this should be in README either since this is not addressing how to run deploy fullerite.

I would go for the approach I recommended earlier which is to automate this in the Makefile
*only* if the user does not have a _vendor directory. That should work.

I also disagree that 99% of our users will not run into this. Since just yesterday 100% of our users ran into this.

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Contributing to fullerite
+
+We welcome all contribution to fullerite, If you have a feature request or you want to improve
+existing functionality of fullerite - it is probably best to open a pull request with your changes.
+
+## Adding new dependency
+
+If you want to add new external dependency to fullerite, please make sure it is added to `Gomfile`.
+Do not forget to specify `TAG` or `commit_id` of external git repository.  More information about
+`Gomfile` can be found from https://github.com/mattn/gom.
```
Comment (tsheasha) [inline contributing.md]: ... can be found *at* ...

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Ensure code is formatted, tested and passes golint.
+
+Running `make` should do all of the above. If you see any failures or errors while running `make`,
+please fix them before opening pull request.
```
Comment (tsheasha) [inline contributing.md]: ... before opening *your | a | the* pull request ...

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## Building and compiling
+
+Running `make` should build fullerite binary and place it in `bin` directory.
```
Comment (tsheasha) [inline contributing.md]: should build *the* fullerite *go* binary .... and place it in *the* bin directory

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+## Building package fails or gom install fails
+
+If you were using older way of vendoring external dependencies you should delete `src/github.com`, `pkg`
```
Comment (tsheasha) [inline contributing.md]: what do you mean by older? maybe we can say If you are using *another* way of vendoring ..

**Comment by poros** [comment]
Since we do not have a "real and official" non-API documentation, my suggestion is to put literally _everything_ in README.md for now. As soon as we set up a real documentation, we'll start to structure the content better.

**Comment by poros** [comment]
lgtm

**Comment by tsheasha** [comment]
+1

**Pull request merged**